src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.cs

```diff
@@ -390,18 +392,33 @@ protected SynthesizedImplementationMethod GenerateIteratorGetEnumerator(MethodSy
             if ((object)initialThreadIdField != null)
             {
                 managedThreadId = MakeCurrentThreadId();
+
+                var thenBuilder = ArrayBuilder<BoundStatement>.GetInstance(4);
+                thenBuilder.Add(
+                    // this.state = {initialState};
+                    F.Assignment(F.Field(F.This(), stateField), F.Literal(initialState)));
+
+                thenBuilder.Add(
+                    // result = this;
+                    F.Assignment(F.Local(resultVariable), F.This()));
+
+                var extraReset = GetExtraResetForIteratorGetEnumerator();
+                if (extraReset != null)
+                {
+                    thenBuilder.Add(extraReset);
+                }
+
+                thenBuilder.Add(
+                    method.IsStatic || method.ThisParameter.Type.IsReferenceType ? // if this is a reference type, no need to copy it since it is not assignable
+                        F.Goto(thisInitialized) : // goto thisInitialized
+                        (BoundStatement)F.StatementList());
```

Comment:
>(BoundStatement)F.StatementList() [](start = 24, length = 33)

Why add an empty statement list?

Reply:
I don't know. That's existing code though, so I'll leave as-is given the deadline.

---
In reply to: [245386255](https://github.com/dotnet/roslyn/pull/31764#discussion_r245386255) [](ancestors = 245386255)

Reply:
I'll fix in next PR

---
In reply to: [245392475](https://github.com/dotnet/roslyn/pull/31764#discussion_r245392475) [](ancestors = 245392475,245386255)